fixtures/build.js

```diff
@@ -22,28 +20,32 @@ window.EXAMPLE_BUILD = {
       }
     },
     "strategy": {
-      "type": "STI",
-      "stiStrategy": {
-        "builderImage": "openshift/ruby-20-centos7:latest",
-        "image": "openshift/ruby-20-centos7:latest"
+      "type": "Source",
+      "sourceStrategy": {
+        "from": {
+          "kind": "DockerImage",
+          "name": "openshift/ruby-20-centos7:latest"
+        },
+        "incremental": true
       }
     },
     "output": {
       "to": {
-        "kind": "ImageStream",
-        "name": "origin-ruby-sample"
-      },
-      "dockerImageReference": "172.30.73.77:5000/test/origin-ruby-sample",
-      "imageTag": "test/origin-ruby-sample",
-      "registry": "172.30.73.77:5000"
-    }
+        "kind": "ImageStreamTag",
+        "name": "origin-ruby-sample:latest"
```

Comment:
"From" can be ImageStream(deprecated), ImageStreamTag, ImageStreamImage, or DockerImage(pullspec)

"To" can be all of the above except ImageStreamImage (imagestreamimages are immutable so it doesn't make sense to push to one)

This code handles an image stream reference field for us, so it should cover all the logic you'd be interested in:
https://github.com/openshift/origin/blob/master/pkg/build/generator/generator.go#L284